File: Java/189.java
class Solution {
    public void rotate(int[] nums, int k) {
        int n = nums.length;
        k = k%n;
        int[] arr = new int[k];
        for(int i=n-k;i<n;i++) arr[i-n+k] = nums[i];
        for(int i=n-1;i>k-1;i--) nums[i] = nums[i-k];
        for(int i=0;i<k;i++) nums[i] = arr[i];
    }
}

File: Python/189.py
class Solution(object):
    def rotate(self, nums, k):
        lst = []
        n = len(nums)
        k = k%n
        for i in range(n-k,n):
            lst.append(nums[i])
        for i in range(n-k-1,-1,-1):
            nums[i+k] = nums[i]
        for i in range(k):
            nums[i] = lst[i]